scripts/smoldyn.py

```diff
@@ -1,57 +1,138 @@
 import numpy as np
 from simulariumio.smoldyn import SmoldynConverter, SmoldynData
-from simulariumio import UnitData, MetaData, DisplayData, DISPLAY_TYPE, ModelMetaData, InputFileData, TrajectoryConverter
-from simulariumio.filters import TranslateFilter
+from simulariumio import (
+    UnitData,
+    MetaData,
+    DisplayData,
+    DISPLAY_TYPE,
+    ModelMetaData,
+    InputFileData,
+    JsonWriter,
+    CameraData,
+)
+
 
 def lambda_handler(event, context):
-    box_size = 100.
+    display_data = None
+    camera_data = None
+    box_size = None
+    model_meta_data = None
+    if "display_data" in event:
+        display_data = dict()
+        for index in event["display_data"]:
+            agent_info = event["display_data"][index]
+            for agent_name in agent_info:
+                agent_data = agent_info[agent_name]
+                display_data[agent_name] = DisplayData(
+                    name=agent_data.get("name"),
+                    radius=float(agent_data.get("radius", 1.0)),
+                    display_type=agent_data.get("display_type", DISPLAY_TYPE.SPHERE),
+                    url=agent_data.get("url"),
+                    color=agent_data.get("color"),
+                )
+    if "meta_data" in event:
+        metadata = event["meta_data"]
+        if "box_size" in metadata and len(metadata["box_size"]) == 3:
+            box_size = np.array(
+                [
+                    float(metadata["box_size"]["0"]),
+                    float(metadata["box_size"]["1"]),
+                    float(metadata["box_size"]["2"]),
+                ]
+            )
 
-    example_data = SmoldynData(
-        meta_data=MetaData(
-            box_size=np.array([box_size, box_size, box_size]),
-            trajectory_title="Some parameter set",
-            model_meta_data=ModelMetaData(
-                title="Some agent-based model",
-                version="8.1",
-                authors="A Modeler",
-                description=(
-                    "An agent-based model run with some parameter set"
+        if "camera_defaults" in metadata:
+            camera_defaults = metadata["camera_defaults"]
+            position = None
+            up_vector = None
+            look_at_position = None
+            fov_degrees = None
+            if "position" in camera_defaults and len(camera_defaults["position"]) == 3:
+                position = np.array(
+                    [
+                        float(camera_defaults["position"]["0"]),
```

Comment:
Is there a chance that a user will input just a single xyz value and not the others? Because as it stands, my code doesn't cover that case